cartographer/cloud/internal/map_builder_server.h

```diff
@@ -69,12 +71,16 @@ class MapBuilderContext : public MapBuilderContextInterface {
 
 class MapBuilderServer : public MapBuilderServerInterface {
  public:
+  using MapBuilderFactory =
+      std::function<std::unique_ptr<mapping::MapBuilderInterface>(
+          const mapping::proto::MapBuilderOptions&,
+          mapping::PoseGraph::GlobalSlamOptimizationCallback)>;
   friend MapBuilderContext<mapping::Submap2D>;
   friend MapBuilderContext<mapping::Submap3D>;
 
   MapBuilderServer(
       const proto::MapBuilderServerOptions& map_builder_server_options,
-      std::unique_ptr<mapping::MapBuilderInterface> map_builder);
+      MapBuilderFactory map_builder_factory);
```

Comment:
This change should be made in a separate commit with a follow up in cartographer_ros